gui/src/app/login/login.component.ts

```diff
@@ -1,12 +1,26 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
+import {UsersService} from "../core/services/users.service";
+import {Router} from "@angular/router";
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss']
 })
-export class LoginComponent {
+export class LoginComponent implements OnInit {
 
-  constructor() { }
+  constructor(private userService: UsersService,
+              private router: Router) {
+  }
+
+  ngOnInit(): void {
+    if (this.userService.getUser() != null) {
```

Comment:
This is not working as expected. The method `getUser()` returns an Observable object, not the user! You have to subscribe to the object and validate the returned result.